processor/filterprocessor/factory.go

```diff
@@ -91,7 +108,15 @@ func createTracesProcessor(
 	cfg component.Config,
 	nextConsumer consumer.Traces,
 ) (processor.Traces, error) {
-	fp, err := newFilterSpansProcessor(set.TelemetrySettings, cfg.(*Config))
+	oCfg := cfg.(*Config)
+	if oCfg.Spans.Include != nil || oCfg.Spans.Exclude != nil {
+		set.Logger.Warn(
+			"The span `include` and `exclude` configuration options have been deprecated and will be removed in 0.74.0. Use `traces.span` instead.",
```

Comment:
nit: Use `::` as a separator when referring to nested options, since this name can then be used literally on configuration
```suggestion
			"The span `include` and `exclude` configuration options have been deprecated and will be removed in 0.74.0. Use `traces::span` instead.",
```